File: lua/problem_71.lua
local M = {}

local function rand7() return math.random(1, 7) end

function M.rand5()
    while true do
        local v = rand7()
        if v <= 5 then return v end
    end
end

function M.problem71()
    print("Problem 71:")
    local counts = {}
    for i = 1, 5 do counts[i] = 0 end
    for _ = 1, 50000 do
        local v = M.rand5()
        assert(v >= 1 and v <= 5)
        counts[v] = counts[v] + 1
    end
    for i = 1, 5 do
        assert(counts[i] > 7000 and counts[i] < 13000, "bad distribution")
    end
    print("All Tests Passed")
end

return M

File: typescript/problem_71.ts
function rand7ts71(): number { return Math.floor(Math.random()*7)+1; }

export function rand5ts71(): number {
    while (true) { const v = rand7ts71(); if (v <= 5) return v; }
}

export function problem71(): void {
    console.log("Problem 71:");
    const counts = new Array(6).fill(0);
    for (let i=0;i<50000;i++) { const v=rand5ts71(); if(v<1||v>5) throw new Error("out of range"); counts[v]++; }
    for (let i=1;i<=5;i++) if(counts[i]<7000||counts[i]>13000) throw new Error("bad distribution");
    console.log("All Tests Passed");
}
